frontend/src/store/root.js

```diff
@@ -27,6 +31,28 @@ const actions = {
       .then(data => commit('REFRESH_COURSES', data))
       .catch(err => console.warn(err))
       .finally(() => commit('TOGGLE_LOADING', false))
+  },
+  getFaculties({ commit, getters }) {
+    // avoid repeats of this
+    if (getters.faculties.length) {
+      return
+    }
+    commit('TOGGLE_LOADING', true)
```

Comment:
Wouldn't even bother with toggling loading. As this has to be loaded at App init. Keep it simple basically.